pkg/etcd-backup-operator/controller/operator.go

```diff
@@ -0,0 +1,64 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/Sirupsen/logrus"
+	v1beta2 "github.com/coreos/etcd-operator/pkg/apis/etcd/v1beta2"
+	"github.com/coreos/etcd-operator/pkg/client"
+	"github.com/coreos/etcd-operator/pkg/generated/clientset/versioned"
+	"github.com/coreos/etcd-operator/pkg/util/constants"
+	"github.com/coreos/etcd-operator/pkg/util/k8sutil"
+
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+type Backup struct {
+	namespace string
+	name      string
+	// k8s workqueue pattern
+	indexer  cache.Indexer
+	informer cache.Controller
+	queue    workqueue.RateLimitingInterface
+
+	kubecli       kubernetes.Interface
+	backupCRCli   versioned.Interface
+	kubeExtClient apiextensionsclient.Interface
+}
+
+// New creates a backup operator.
+func New() *Backup {
+	return &Backup{
+		namespace:     os.Getenv(constants.EnvOperatorPodNamespace),
+		name:          os.Getenv(constants.EnvOperatorPodName),
+		kubecli:       k8sutil.MustNewKubeClient(),
+		backupCRCli:   client.MustNewInCluster(),
+		kubeExtClient: k8sutil.MustNewKubeExtClient(),
+	}
+}
+
+// Start starts the Backup operator.
+func (b *Backup) Start(ctx context.Context) error {
+	err := b.init(ctx)
+	if err != nil {
+		return err
+	}
+	go b.run(ctx)
+	<-ctx.Done()
+	return ctx.Err()
+}
+
+func (b *Backup) init(ctx context.Context) error {
+	logrus.Fatalf("checking if crd %v exists", v1beta2.EtcdBackupCRDName)
```

Comment:
What is this for?

Reply:
will remove since checking crd is not necessary